main.go

```diff
@@ -462,3 +487,207 @@ func stopStalePipelines (client *gitlab.Client, vars []*gitlab.PipelineVariable)
 
 	}
 }
+
+// syncIfOSHasEnterpriseRepo detects whether a commit has been merged to
+// the Open Source edition of a repo, and then creates a PR-branch
+// in the Enterprise edition, which is then used in order to open
+// a PR to the Enterprise repo with the OS changes.
+func syncIfOSHasEnterpriseRepo(conf *config, gpr *github.PullRequestEvent) error {
+
+	repo := gpr.GetRepo()
+	if repo == nil {
+		return fmt.Errorf("syncIfOSHasEnterpriseRepo: Failed to get the repository information")
+	}
+
+	// Enterprise repo sentinel
+	switch repo.GetName() {
+	case "deployments":
+	case "useradm":
+	default:
+		log.Debugf("syncIfOSHasEnterpriseRepo: Non-Enterprise repository detected: (%s). Not syncing", repo.GetName())
+		return nil
+	}
+
+	pr := gpr.GetPullRequest()
+	if pr == nil {
+		return fmt.Errorf("syncIfOSHasEnterpriseRepo: Failed to get the pull request")
+	}
+
+	// If the action is "closed" and the "merged" key is "true", the pull request was merged.
+	// While webhooks are also triggered when a pull request is synchronized, Events API timelines
+	// don't include pull request events with the "synchronize" action.
+	if gpr.GetAction() == "closed" && pr.GetMerged() {
+
+		// Only sync on Merges to master or release branches and
+		// verify release branches.
+		branch := pr.GetBase()
+		if branch == nil {
+			return fmt.Errorf("syncIfOSHasEnterpriseRepo: Failed to get the base-branch of the PR: %v", branch)
+		}
+
+		syncBranches := regexp.MustCompile(`(master|[0-9]{1,2}\.[0-9]{1,2}\.x)`)
+		branchRef := branch.GetRef()
+		if !syncBranches.MatchString(branchRef) {
+
+			log.Debugf("syncIfOSHasEnterpriseRepo: Detected a merge into another branch than master or a relase branch: (%s), no syncing done", branchRef)
+
+		} else {
+
+			log.Infof("syncIfOSHasEnterpriseRepo: Merge to (%s) in an OS repository detected. Syncing the repositories...", branchRef)
+
+			PRBranchName := "mergeostoent_" + strconv.Itoa(pr.GetNumber())
```

Comment:
:+1:

Reply:
Good, I take it the naming is approved then.

Reply:
Yes. Your emoji interpreting skills are good.

Reply:
haha, I'm improving xD